jwst/jump/jump.py

```diff
@@ -75,12 +89,43 @@ def detect_jumps (input_model, gain_model, readnoise_model,
     # Apply the 2-point difference method as a first pass
     log.info('Executing two-point difference method')
     start = time.time()
-
-    median_slopes = twopt.find_crs(data, gdq, readnoise_2d,
-                                           rejection_threshold, nframes)
-
-    elapsed = time.time() - start
-    log.debug('Elapsed time = %g sec' %elapsed)
+    scisize = data.shape
+    numx = int(scisize[3])
+    numy = int(scisize[2])
+    median_slopes = np.zeros((numy, numx), dtype=np.float32)
+    yincrement = int(numy / numslices)
+    slices = []
+    for i in range(numslices - 1):
+        slices.insert(i, (data[:, :, i * yincrement:(i + 1) * yincrement, :],
+                          gdq[:, :, i * yincrement:(i + 1) * yincrement, :],
+                          readnoise_2d[i * yincrement:(i + 1) * yincrement, :],
+                          rejection_threshold, nframes))
+    slices.insert(numslices - 1, (data[:, :, (numslices - 1) * yincrement:numy, :],  # last slice get the rest
+                                 gdq[:, :, (numslices - 1) * yincrement:numy, :],
+                                 readnoise_2d[(numslices - 1) * yincrement:numy, :],
+                                 rejection_threshold, nframes))
+    if yincrement * numslices != numy:  # last slice is a larger one
+        print("odd last slice ", yincrement, numy - (numslices - 1) * yincrement)
```

Comment:
Shouldn't have a print statement in operational code. Also, I think that all of the boilerplate for creating and distributing slices to cores, etc. could use a little more commentary, for the sake of future developers.